### PR TITLE
feat: use ethers v6

### PR DESCRIPTION
**Pull request by janek26**
### Issue / feature description

There was a major ethers release (v6) with native BigInt and man more nice things.

### Checklist

- [x] Rebased to the last commit of the target branch (or merged)
- [x] Code self-reviewed
- [x] Code self-tested
- [x] Tests updated (if needed)
- [x] All tests are passing locally

Please keep your pull request as small as possible. If you need to make multiple changes, please create separate pull requests for each change. Create a draft pull request if you need early feedback. This will mark the pull request as a work in progress and prevent it from being reviewed or merged until you are ready.

Please move drafts to the ready for review (regular PR) when you are ready to start the review process and other developers will pick it up from there.

### CODE REVIEW

File: packages/extension/src/background/wallet.ts [hunk 1]
```diff
@@ -132,7 +136,7 @@ export class Wallet {
     const ethersWallet = ethers.Wallet.createRandom()
     const encryptedBackup = await ethersWallet.encrypt(
       password,
-      { scrypt: { N: SCRYPT_N } },
+      // { scrypt: { N: SCRYPT_N } },
```
Comment (diegodelrieu) [inline packages/extension/src/background/wallet.ts:139]: Any reason to leave this commented ?

**Review by diegodelrieu — APPROVED** [review]

**Comment by simonheys** [comment]
Very nice! Seeing an error serialising bigint with JSON.stringify here:

https://github.com/argentlabs/argent-x/blob/develop/packages/extension/src/ui/services/swr.ts#L23

`TypeError: Do not know how to serialize a BigInt`

Some various solutions suggested here, either adding a 'replacer' that stringifies the bigint or ... patching bigint ... https://github.com/GoogleChromeLabs/jsbi/issues/30

File: packages/extension/src/background/wallet.ts (hunk 1 above)
Comment (janek26) [reply packages/extension/src/background/wallet.ts:139]: good catch, thanks

**Comment by janek26** [comment]
> Very nice! Seeing an error serialising bigint with JSON.stringify here:
> 
> https://github.com/argentlabs/argent-x/blob/develop/packages/extension/src/ui/services/swr.ts#L23
> 
> `TypeError: Do not know how to serialize a BigInt`
> 
> Some various solutions suggested here, either adding a 'replacer' that stringifies the bigint or ... patching bigint ... [GoogleChromeLabs/jsbi#30](https://github.com/GoogleChromeLabs/jsbi/issues/30)

thanks, updated!

**Review by simonheys — APPROVED** [review]
Works good now!

**Pull request merged**